bots/push-rewrite

```diff
@@ -30,6 +30,13 @@ def execute(*args):
 def git(*args):
     return execute("git", *args).strip()
 
+def find_pr_with_sha(api, sha):
+    pulls = api.pulls()
+    for pull in pulls:
+        if pull["head"]["sha"] == sha:
+            return pull
```

Comment:
Too bad that the [commits API](https://api.github.com/repos/cockpit-project/cockpit/commits/a935533b4afae934dad4d8cb13a9ecc4bfdf49c9) doesn't refer back to the PR.. So I suppose this is necessary. But this should have some error handling if the PR isn't found.